scripts/wsrep_sst_rsync.sh

```diff
@@ -69,18 +70,42 @@ check_pid_and_port()
 
     local port_info=$(lsof -i :$rsync_port -Pn 2>/dev/null | \
         grep "(LISTEN)")
+    local is_listening_all=$(echo $port_info | \
+        grep "*:$rsync_port" 2>/dev/null)
+    local is_listening_addr=$(echo $port_info | \
+        grep "$rsync_addr:$rsync_port" 2>/dev/null)
     local is_rsync=$(echo $port_info | \
         grep -w '^rsync[[:space:]]\+'"$rsync_pid" 2>/dev/null)
 
-    if [ -n "$port_info" -a -z "$is_rsync" ]; then
-        wsrep_log_error "rsync daemon port '$rsync_port' has been taken"
-        exit 16 # EBUSY
+    if [ ! -z "$is_listening_all" -o ! -z "$is_listening_addr" ]; then
+        if [ -z "$is_rsync" ]; then
+            wsrep_log_error "rsync daemon port '$rsync_port' has been taken"
+            exit 16 # EBUSY
+        fi
     fi
     check_pid $pid_file && \
         [ -n "$port_info" ] && [ -n "$is_rsync" ] && \
         [ $(cat $pid_file) -eq $rsync_pid ]
 }
 
+is_local_ip()
+{
+  local address="$1"
+  local wherebin=`which ifconfig`
+  if [ -z "$wherebin" ]
+  then
+    wherebin=`which ip`
+    wherebin="$wherebin address show"
+    # Add an slash at the end, so we don't get false positive : 172.18.0.4 matches 172.18.0.41
+    address="${address}/"
+  else
+    # Add an space at the end, so we don't get false positive : 172.18.0.4 matches 172.18.0.41
+    address="$address "
```

Comment:
Could you also add a comment to explain why do we need 2 different address terminators?